packages/dd-trace/test/telemetry/dependencies.spec.js

```diff
@@ -13,6 +13,15 @@ const moduleLoadStartChannel = dc.channel('dd-trace:moduleLoadStart')
 const originalSetImmediate = global.setImmediate
 
 describe('dependencies', () => {
+  function setImmediate2 (callback, ...args) {
+    return callback(...args)
+  }
+
+  setImmediate2.__promisify__ = function (...args) {
+    setImmediate2(() => undefined, ...args)
+    return new Promise((resolve) => resolve(undefined))
```

Comment:
And here:

```suggestion
    return /** @type {Promise<void>} */ (new Promise((resolve) => resolve()))
```

But maybe even simpler:

```suggestion
    return Promise.resolve()
```